### ISSUE TITLE
Add validation split code to code-behind

### ISSUE THREAD

**Issue by luisquintanilla** [dotnet/machinelearning-modelbuilder#2555]
Currently you're able to choose your data split strategy in advanced data options. 

Add code to the code-behind files to match selection made during training.

**Comment by luisquintanilla** [comment]
@zewditu here's my first attempt at a method name. We can iterate on it. 

`DatasetTrainTestSplit`

**Comment by luisquintanilla** [comment]
The reason why we're not using the dataset for tuning hyperparameters is in the code-behind `BuildPipeline`, users get the actual pipeline. Not the AutoML code to retrain their model. As a result, it's not 1:1 with what Model Builder is doing. Therefore, these are some potential options:

- In `RetrainModel`, perform the split and do the training only on the training portion. Then, in the *\*.evaluate.cs* file, use the test dataset to evaluate the performance of the model. 
- For `BuildPipeline`, provide the AutoML training code which uses the validation dataset for hyperparameter tuning instead of the actual ML.NET transforms and pipeline.
- Create a new method `RetrainWithAutoML` which uses the validation dataset for hyperparameter tuning.

**Comment by luisquintanilla** [comment]
@luisquintanilla to add additional details for what should happen depending on the split (CV vs TrainTestSplit)

**Comment by luisquintanilla** [comment]
Next steps:

- Use seed & new MLContext in \*.training.cs and \*.evaluate.cs files to guarantee the splits are the same for training and evaluation operations.
- For CV, automatically perform the split and run training on training split and evaluate on validation / test set.
